packages/backend/src/controllers/sqlRunnerController.ts

```diff
@@ -188,6 +189,40 @@ export class SqlRunnerController extends BaseController {
         };
     }
 
+    /**
+     * Gets chart and schedules a job to get its results
+     * @param projectUuid - the uuid of the project
+     * @param uuid - the uuid of the saved chart
+     * @param req - express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('saved/{uuid}/chart-and-results')
+    @OperationId('getSavedSqlChartWithResults')
+    async getSavedSqlChartWithResults(
+        @Path() uuid: string,
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiSqlChartWithResults> {
+        this.setStatus(200);
+
+        const chart = await this.services
```

Comment:
one `minimal` performance optimization. Since the `job` is going to take longer, perhaps we want to do `scheduleSqlJob` first. Or even do both in parallel with Promise.all